src/core/MoveValidator.ts

```diff
@@ -19,13 +20,15 @@ class MoveValidator {
     if (obstaclePiece) {
       if (obstaclePiece.color === piece.color) {
         return false;
+      } else {
+        move.isCapture = true;
```

Comment:
To jest lekki side effect, ale nie wiem czy na tę chwilę można to zrobić lepiej, imho może zostać, bo działa :P.

Reply:
W pełni się zgadzam, pomyślimy nad bardziej eleganckim rozwiązaniem w przyszłości